uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-day.component.js

```diff
@@ -124,40 +74,20 @@ define([
       LeavePopup.openModal(leaveRequest, leaveType, selectedContactId, isSelfRecord);
     }
 
-    /**
-     * Copies dates from Leave Request, converts them from String
-     * to Date type and sets to the component.
-     */
-    function resolveDayDates () {
-      vm.dates.from = new Date(vm.contactData.leaveRequest.from_date);
-      vm.dates.to = new Date(vm.contactData.leaveRequest.to_date);
-    }
-
-    /**
-     * Determines the label for the day depending on the calculation unit or if
-     * it's an accrued TOIL request.
-     */
-    function resolveDayLabel () {
-      if (vm.contactData.isAccruedTOIL) {
-        vm.label = 'AT';
-      } else if (vm.calculationUnit.name === 'days') {
-        resolveDayLabelForDaysCalculationUnit();
-      } else {
-        resolveDayLabelForHoursCalculationUnit();
-      }
-    }
-
     /**
      * Determines the label for the day when calculation units are set to days.
      *
      * AM: leave requests for half day AM
      * PM: leave requests for half day PM
      * Otherwise leave empty.
+     *
+     * @param  {Object} leaveRequestAttributes
+     * @return {String}
      */
-    function resolveDayLabelForDaysCalculationUnit () {
-      vm.label = vm.contactData.isAM ? 'AM'
-        : vm.contactData.isPM ? 'PM'
-        : '';
+    function resolveDayLabelForDaysCalculationUnit (leaveRequestAttributes) {
+      return leaveRequestAttributes.isAM ? 'AM'
+        : leaveRequestAttributes.isPM ? 'PM'
+          : '';
```

Comment:
is it really requesting these 2 spaces? (just asking)

Reply:
I think the confusion happens because it should be like that instead:

```js
return leaveRequestAttributes.isAM
  ? 'AM'
  : leaveRequestAttributes.isPM
    ? 'PM'
    : '';
```
Hope this is clearer ^